ark/api/v1alpha1/agent_types.go

```diff
@@ -15,6 +15,16 @@ type ToolFunction struct {
 	Value string `json:"value,omitempty"`
 }
 
+type ToolPartial struct {
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:MinLength=1
+	Name string `json:"name,omitempty"`
```

Comment:
Could we add a description comment for all fields going forwards, CF

https://github.com/mckinsey/agents-at-scale-ark/pull/310/files#diff-e1044445ef2cf3d75a9a9d9c6621df70c3db88373e765df344bbc604de141debR74